### ISSUE TITLE
Apple Share Sheet Support?

### ISSUE THREAD

**Issue by n3bulous** [obsidianmd/obsidian-clipper#262]
I didn’t see it on the roadmap, but sharing via the share sheet would be immensely easier and usable from not web browsers. 

I’m sure Apple makes this 10x harder than it should be…

**Comment by kepano** [comment]
You can already share using the share sheet on Safari. Just click the down arrow, or the share icon if you're on iOS.

<img width="263" alt="Screenshot 2024-12-14 at 10 29 26" src="https://github.com/user-attachments/assets/73dbc749-b84f-41fc-9b1f-b8e16fdcc472" />

**Comment by cole-h** [comment]
I think the original issue is more about not needing to use a Safari browser addon and instead being able to use an app share target in the share sheet, like so:

![image](https://github.com/user-attachments/assets/55cbdd7b-442a-46e8-8c6f-fe2703d53286)

(I do understand that it would have to be a separate app in the share sheet -- I'm fine with this)

Rather than having to go through the relatively cumbersome process of tapping the "addons" button on the side of the URL box and picking the Obsidian Web Clipper addon, then tapping "Add to Obsidian", then tapping "Open" to open it in Obsidian, then tapping "Allow Paste" from Safari.

Basically what I selfishly want is for the flow to be:

1. Share to Web Clipper in share sheet
2. Replace the "Open in Obsidian" button with the share button that will open up a share sheet with the text to add to Obsidian

That would make it much easier and quicker to add a clipping to my vault, no need to dance around iOS asking for permission to open Obsidian and then asking for permission to paste the clipping's contents.

**Comment by calebhearth** [comment]
Agreed with @cole-h. Is it possible to have that use-case considered?

Specifically my pain point right now is that in the in-app browsers included in a lot of apps (Ivory, GitHub, Reeder, etc.) there is not an extensions menu, so clipping requires:

1. Finding content you want in these apps
2. Opening in the in-app browser
    1. Optional but likely: Forgetting that this feature is unsupported and hitting the share sheet 🤣
3. Re-opening in Safari
4. Hitting the extension menu
5. Fighting with the extension menu since it just toggled Reader mode instead of opening
6. Scrolling to find Obsidian Web Clipper
7. Hitting Add to Obsidian

I'd love to have some of that friction removed if possible.

**Comment by kepano** [comment]
It seems like the feature request is for Apple to add extensions to the in-app browser view.

**Comment by calebhearth** [comment]
My feature request is for the clipper to add a Share Extension that performs a similar function to the current flow, but from the system share sheet. I believe that's similar to what was originally requested based on the screenshot.

https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/Share.html

**Comment by kepano** [comment]
The problem is that share sheets do not have access to the page DOM so it would not be able to do much.

**Comment by calebhearth** [comment]
That's a good point. I think `NSExtensionJavaScriptPreprocessingFile` helps with this but I'm AFK this weekend and can't confirm.

This Hacking With Swift [JavaScript Injection](https://www.hackingwithswift.com/read/19/) series seems to indicate that by step 5 you can retrieve values in Swift from JavaScript injected into the page and defined in step 4 that is part of a Share extension.

Do you think that gets closer to the mark?

**Comment by kevinschweikert** [comment]
Just adding my 5 cents. Now that Pocket shut down i switched to the Web Clipper and that works flawlessly (thank you!). But often i want to add some links from other apps like:

- YouTube
- Bluesky
- Octal (Hackernews Client)
- Reddit
- ....

without opening them in Safari like @calebhearth said. It would be amazing to use the Web Clipper from the native iOS share sheet
